MOxUnit/@MOxUnitTestResult/report_xml.m

```diff
@@ -0,0 +1,164 @@
+            x = [x, ' line="' num2str(es.line) '"' ...
+                    '>' ...
+                    '<system-err message="' ...
+                        get_first_line(reason.message) ...
+                        '">' ...
+                    reason.message ...
+                    '</system-err>' ...
+                    '</testcase>' ...
+                    ];
+
+        case 'f'
+            % Fill the contents for a failure
+            es = reason.stack(1);
+            reason.message = strip_xml(reason.message);
+            x = [x, ' line="' num2str(es.line) '"' ...
+                    '>' ...
+                    '<failure message="' ...
+                        get_first_line(reason.message) ...
+                        '">' ...
+                    reason.message ...
+                    '</failure>' ...
+                    '</testcase>' ...
+                    ];
+
+        case 's'
+            % Fill the contents for a skipped test
+            x = [x, '>' ...
+                    '<skipped message="' reason '" />' ...
+                    '</testcase>' ...
+                    ];
+	
+        case '.'
+            % There are no contents for a success, so we just close
+            % the tag
+            x = [x, ' />'];
+	
+        otherwise
+            error('moxunit:notImplemented', ...
+                  'Unrecognised test case result code: %s', lower(c(1)));
+
+    end
+
+
+function string=strip_xml(string)
+% Strip XML tags from a string
+%
+% string=strip_xml(string)
+%
+% Inputs:
+%   string          A string which may contain XML tags Any '<' and '>'
+%                   characters which are not part of an XML tag should
+%                   be escaped as '\<' and '\>', otherwise they will be
+%                   removed.
+%
+% Output:
+%   string          The string with any XML tags removed. Any escaped
+%                   characters ('\<', '\>') remain as they are - still
+%                   escaped.
+%
+% SCL 2015
+
+    % We check to make sure the character before the '<' isn't a '\' or
+    % the start of the string. If it isn't, we capture until the next
+    % non-escaped '>' character and remove the whole chunk. However,
+    % we need to include the leading non-'\' character (the first group)
+    % in the replacement output.
+    string = regexprep(string , '(^|[^\\])(<.*?[^\\]>)', '$1');
+
+
+function string=get_first_line(string)
+% Get the first line (before newline character) of a string
+%
+% string=get_first_line(string)
+%
+% Inputs:
+%   string          A string, possibly multiline.
+%
+% Output:
+%   string          The string with any XML tags removed. Any escaped
+%                   characters ('\<', '\>') remain as they are - still
+%                   escaped.
+%
+% SCL 2015
+
+    % Trim off any newlines from the start of the string
+    % NB: sprintf('\n')==10
+    string = strtrim(string);
+    start_index = find(string(2:end)~=10, 1);
```

Comment:
[Unicode](https://en.wikipedia.org/wiki/Newline#Unicode) always has LF encoded as 10 in decimal.
So comparing with 10 will be the same as using sprintf('\n') on all systems. I do things this way because `find(_, 1)` is faster than `strfind` --- strfind can't stop after 1.
> LF:    Line Feed, U+000A